packages/jsii/test/compiler.test.ts

```diff
@@ -103,6 +111,43 @@ describe(Compiler, () => {
       await remove(sourceDir);
     }
   });
+
+  test('rootDir is added to assembly', async () => {
+    const outDir = 'jsii-outdir';
+    const rootDir = 'jsii-rootdir';
+    const sourceDir = await mkdtemp(join(tmpdir(), 'jsii-tmpdir'));
+    await ensureDir(join(sourceDir, rootDir));
+
+    try {
+      await writeFile(
+        join(sourceDir, rootDir, 'index.ts'),
+        'export class MarkerA {}',
+      );
+      // Intentionally using lower case name - it should be case-insensitive
+      await writeFile(join(sourceDir, rootDir, 'readme.md'), '# Test Package');
+
+      const compiler = new Compiler({
+        projectInfo: {
+          ..._makeProjectInfo(sourceDir, join(outDir, 'index.d.ts')),
+          tsc: {
+            outDir,
+            rootDir,
+          },
+        },
+        failOnWarnings: true,
+        projectReferences: false,
+      });
+
+      await compiler.emit();
+
+      const assembly = await readJson(join(sourceDir, '.jsii'), 'utf-8');
+      expect(assembly.metadata).toEqual({
```

Comment:
For robustness against future additions to metadata:

```suggestion
      expect(assembly.metadata).toEqual(expect.objectContaining({
```